database/sqlite/sqlite_health.c

```diff
@@ -1352,129 +1353,73 @@ void sql_health_alarm_log2json(RRDHOST *host, BUFFER *wb, time_t after, const ch
-                 "-" :
-                 format_value_and_unit(
-                     old_value_string, 100, sqlite3_column_double(stmt_query, 24), (char *)sqlite3_column_text(stmt_query, 17), -1),
-             (long unsigned int)sqlite3_column_int64(stmt_query, 25),
-             (sqlite3_column_int64(stmt_query, 9) & HEALTH_ENTRY_FLAG_SILENCED) ? "true" : "false");
-
-         health_string2json(wb, "\t\t", "summary", (char *)sqlite3_column_text(stmt_query, 31), ",\n");
-         health_string2json(wb, "\t\t", "info", (char *)sqlite3_column_text(stmt_query, 18), ",\n");
-
-         if (unlikely(sqlite3_column_int64(stmt_query, 9) & HEALTH_ENTRY_FLAG_NO_CLEAR_NOTIFICATION)) {
-            buffer_strcat(wb, "\t\t\"no_clear_notification\": true,\n");
-         }
-
-         buffer_strcat(wb, "\t\t\"value\":");
-         if (sqlite3_column_type(stmt_query, 23) == SQLITE_NULL)
-            buffer_strcat(wb, "null");
-         else
-            buffer_print_netdata_double(wb, sqlite3_column_double(stmt_query, 23));
-         buffer_strcat(wb, ",\n");
-
-         buffer_strcat(wb, "\t\t\"old_value\":");
-         if (sqlite3_column_type(stmt_query, 24) == SQLITE_NULL)
-            buffer_strcat(wb, "null");
-         else
-            buffer_print_netdata_double(wb, sqlite3_column_double(stmt_query, 24));
-         buffer_strcat(wb, "\n");
-
-         buffer_strcat(wb, "\t}");
-
-         freez(edit_command);
-     }
+        buffer_json_add_array_item_object(wb); // this node
+
+        buffer_json_member_add_string_or_empty(wb, "hostname",             rrdhost_hostname(host));
+                  buffer_json_member_add_int64(wb, "utc_offset",           (int64_t)host->utc_offset);
+        buffer_json_member_add_string_or_empty(wb, "timezone",             rrdhost_abbrev_timezone(host));
+                  buffer_json_member_add_int64(wb, "unique_id",            (int64_t) sqlite3_column_int64(stmt_query, 0));
+                  buffer_json_member_add_int64(wb, "alarm_id",             (int64_t) sqlite3_column_int64(stmt_query, 1));
+                  buffer_json_member_add_int64(wb, "alarm_event_id",       (int64_t) sqlite3_column_int64(stmt_query, 2));
+        buffer_json_member_add_string_or_empty(wb, "config_hash_id",       config_hash_id);
+        buffer_json_member_add_string_or_empty(wb, "transition_id",        transition_id);
+        buffer_json_member_add_string_or_empty(wb, "name",                 (const char *) sqlite3_column_text(stmt_query, 12));
+        buffer_json_member_add_string_or_empty(wb, "chart",                (const char *) sqlite3_column_text(stmt_query, 13));
+        buffer_json_member_add_string_or_empty(wb, "context",              (const char *) sqlite3_column_text(stmt_query, 29));
+        buffer_json_member_add_string_or_empty(wb, "class",                sqlite3_column_text(stmt_query, 26) ? (const char *) sqlite3_column_text(stmt_query, 26) : (char *) "Unknown");
```

Comment:
You are using `Unknown` different times, probably it is better to set it as a macro helping old compilers to optimize the code.